src/back/TicketManager/src/Core/TicketManager.Core.Services/Endpoints/Events/CreateEventEndpoint.cs

```diff
@@ -29,10 +31,17 @@ public override async Task HandleAsync(CreateEventRequest req, CancellationToken
             req.Name,
             req.Description,
             req.Location,
-            req.Date,
-            req.Sectors
-                .Select(s => new Event.SectorData(s.Name, s.PriceInSmallestUnit, s.NumberOfColumns, s.NumberOfRows))
-                .ToList());
+            req.Date);
+
+        foreach (var sector in req.Sectors)
+        {
+            sectors.Add(new Sector(
+                @event.Id, 
+                sector.Name, 
+                sector.PriceInSmallestUnit,
+                sector.NumberOfColumns,
+                sector.NumberOfRows));
+        }
```

Comment:
For now it can stay, but when we integrate our system with message bus we have to change it.
The problem here is that we operate on multiple aggregates in single transaction what we shouldn't do. The purposes of aggregates is to define consistency boundaries. Aggregate is meant to be consistent inside, but relations between aggregates are eventually consistent, because keeping all system consistent in every second is really hard process and in big systems it's even impossible. Of course our system is not big and it will be used by three people at most, but we build it as big system for learning purposes.

Reply:
Yes, but it comes to the point that you wanted `Sector`s to not be a part of `Event` when they are strictly connected. How would like me to add only `Event` and ignore adding `Sector`s? They are after all connected and `Sector`s can't exist on their own

Reply:
You're thinking about it in a wrong way. Yes that's true that they are connected to each other.

In big systems there are many things that are connected to each other. Keeping theme consistent in every moment of system lifetime forces you to do db transactions that operates on large number of objects. And that causes performance and maintaining issues. To avoid those issues you have to introduce eventual consistency in some parts of the system.

Our system is not that big that it doesn't work without eventual consistency, but we want to introduce eventual consistency for learning purposes.

Also in systems there are parts that have to be consistent all the time, because otherwise system doesn't work. But not everything can be consistent all the time. Some parts have to be eventually consistent.

To introduce consistency boundaries we use DDD (Domain Driven Design) which gives us Aggregates. Aggregates are consistency boundaries. Aggregate is consistent inside. But relations between aggregates are eventually consistent. Because of that we shouldn't operate on multiple aggregates in single transaction.

Reply:
Isn't it a little over the top to use MassTransit in this project?
<p color="#ffffff10">also, hi</p>